daemon/k8s_watcher.go

```diff
@@ -258,6 +259,7 @@ func (d *Daemon) EnableK8sWatcher(reSyncPeriod time.Duration) error {
 		reSyncPeriod,
 		cache.ResourceEventHandlerFuncs{
 			AddFunc: func(obj interface{}) {
+				metrics.SetTSValue(metrics.LastK8sEventTS, time.Now())
```

Comment:
I'm not terribly happy adding this to each and every one of these functions. I'm not sure if there is a better place for it. It can go in the actual handler function but it seemed best to do it closest to the event it captured (if we can't copy the objects we still want to record that we saw an event).

Reply:
I think that's fine. We may have a generic controller soon to handle all events where this could be moved to eventually.